lib/logstash/inputs/kafka.rb

```diff
@@ -270,8 +276,15 @@ def thread_runner(logstash_queue, consumer)
           end
         end
       rescue org.apache.kafka.common.errors.WakeupException => e
-        raise e if !stop?
+        unless stop?
+          logger.error("wakeup exception in consumer #{consumer_identifier}: #{e}")
+          raise e
+        end
+      rescue => e
+        logger.error("uncaught exception in consumer #{consumer_identifier}: #{e}")
```

Comment:
Can we use structured logging in these places? Logstash loggers let you say:

`logger.error("uncaught exception in consumer", :consumer_identifier => consumer_identifier, :message => e.message, :error_class => e.class.name, :error_backtrace => e.backtrace)`